versions.props

```diff
@@ -1,5 +1,5 @@
 com.fasterxml.jackson.*:* = 2.11.3
-com.github.ben-manes.caffeine:caffeine = 2.9.0
+com.github.ben-manes.caffeine:caffeine = 3.0.0
```

Comment:
[caffeine 3.0 requires JDK 11+](https://github.com/ben-manes/caffeine/releases/tag/v3.0.0) so we can't bump this until we bump tritium's minimum JDK dependency to 11 as well. I think we probably want to pin to 2.9.0 for now, but if downstream consumers start using caffeine 3.x, we may encounter sad times.

```
error: An unhandled exception was thrown by the Error Prone static analysis plugin.
     Please report this at https://github.com/google/error-prone/issues/new and include the following:
  
     error-prone version: 2.5.1
     BugPattern: (see stack trace)
     Stack Trace:
     java.lang.UnsupportedClassVersionError: com/github/benmanes/caffeine/cache/Caffeine has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```
```suggestion
# dependency-upgrader:OFF
# caffeine 3.0 requires JDK 11+ see https://github.com/ben-manes/caffeine/releases/tag/v3.0.0
com.github.ben-manes.caffeine:caffeine = 2.9.0
# dependency-upgrader:ON
```

Reply:
The compatibility should be pretty good. Most of them are interface changes only caffeine implemented, but by semver it’s listed as an api break. I think actual issues will be rare.

Reply:
Excellent, thanks for the heads up @ben-manes and appreciate your hard work on caffeine, its excellent! 

My main concern here is if tritium bumps to caffeine 3.0 it will implicitly flow to all downstream consumers, so binary compatibility breaks (if any) would only be triggered downstream at runtime. Tritium's use of caffeine API surface is pretty small as we're just exposing the cache stats metrics and the only deprecated method we hit in 2.9.0 release https://github.com/palantir/tritium/pull/996 was simple constructor to factory method in test code, so the likelihood for problems seems low, especially based on your diligence with caffeine APIs.

For now, I think I'm going to pin tritium to caffeine 2.9.0 while we test out 3.0.0 downstream (https://github.com/palantir/tritium/pull/1001). There's a few internal things I need to do to prep a tritium 1.x release that requires JDK 11+ and at that point tritium taking the caffeine 3.x bump will be fine.

Reply:
Thanks. Your usages look very benign and should be compatible going forward. I'll continue to support 2.x so there is no rush to upgrade JDKs. The migration to VarHandles forced Caffeine to update its minimum JDK, but I think most projects can move at their own pace and not be impacted if they keep their dependencies up to date and resolve deprecation warnings. Hopefully, though, maybe this year the general ecosystem will make the push to move off of JDK8 (2014).